### PR TITLE
The Fuck plugin (https://github.com/nvbn/thefuck)

### PR DESCRIPTION
**Pull request by black-roland**
[The Fuck](https://github.com/nvbn/thefuck) plugin — magnificent app which corrects your previous console command.

Press `ESC` twice to correct previous console command.

### CODE REVIEW

**Comment by apjanke** [comment]
Dupe of/conflicts with #3861.

@black-roland, @cristim, can you and other users of this come to a consensus on which to use?

**Comment by apjanke** [comment]
@robbyrussell et al, is there an OMZ policy on things named "fuck" and the like? Previous PR #2555 requesting a "fuck" alias was withdrawn, but no comments on why.

**Comment by apjanke** [comment]
Note that the `Esc`-`Esc` key binding conflicts with the `sudo` plugin, which [binds it to adding `sudo` to previous command](https://github.com/robbyrussell/oh-my-zsh/blob/a207a38d634cc10441636bc4359cd8a18c502dea/plugins/sudo/sudo.plugin.zsh#L21). Introduced in #1947.

**Comment by black-roland** [comment]
> Dupe of/conflicts with #3861.

I think these plugins can be combined into one. If @cristim agree, I will add his changes into my plugin.

**Comment by black-roland** [comment]
> Note that the Esc-Esc key binding conflicts with the sudo plugin, which binds it to adding sudo to previous command. Introduced in #1947.

The Fuck partially can replace sudo plugin. The Fuck can prefix with sudo commands like apt-get, yum, etc.

**Comment by robbyrussell** [comment]
I don't have a fucking problem with the usage of the word fuck.

**Comment by apjanke** [comment]
Cool. I think that note in the README is sufficient to address the binding conflict then.

**Comment by black-roland** [comment]
Combined with plugin by @cristim. Also fixed compability with latest The Fuck version.

**Comment by cristim** [comment]
Looks good to me.

File: plugins/thefuck/thefuck.plugin.zsh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+if [[ -z $commands[thefuck] ]]; then
+    echo 'thefuck is not installed, you should "pip install thefuck" first'
+    return -1
+fi
+
+# Register alias
+eval "$(thefuck-alias)"
+
+fuck-command-line() {
+    FUCK="$(THEFUCK_REQUIRE_CONFIRMATION=0 thefuck $(fc -ln -1 | tail -n 1) 2> /dev/null)"
```
Comment (apjanke) [inline plugins/thefuck/thefuck.plugin.zsh]: Could this `$FUCK` be made local? Or is it intended to stick around for reference?

**Comment by apjanke** [comment]
Looks good to me too. Tested on OS X 10.9.5.

File: plugins/thefuck/thefuck.plugin.zsh (hunk 1 above)
Comment (black-roland) [inline plugins/thefuck/thefuck.plugin.zsh]: > Could this $FUCK be made local? Or is it intended to stick around for reference?

Ooops. It's a bug. Added `local`.

**Pull request merged**